src/lib/io/sol-iio.c

```diff
@@ -973,6 +994,9 @@ sol_iio_add_channel(struct sol_iio_device *device, const char *name, const struc
     r = sol_ptr_vector_append(&channel->device->channels, channel);
     SOL_INT_CHECK_GOTO(r, < 0, error);
 
+    /* Some device firmware/hardware feature default state is off, thus need to activate it before read */
+    set_channel_enabled(device, name, true);
```

Comment:
Hi @laykuanloon ,

don't you think it could be useful to log case set_channel_enabled fail (return false)?